druid/src/text/movement.rs

```diff
@@ -51,8 +55,11 @@ pub fn movement(m: Movement, s: Selection, text: &impl EditableText, modify: boo
             }
         }
 
-        Movement::LeftOfLine => 0,
-        Movement::RightOfLine => text.len(),
+        Movement::LeftOfLine => text.left_end_of_line(s.end),
+        Movement::RightOfLine => text.right_end_of_line(s.end),
```

Comment:
These are also the wrong names, given the implementation. Ideally these *would* be the names and we would check writing direction on the paragraph, but we don't have that ability right now.

I think it's fine to leave this as is in this PR, and I'll open an issue describing the problem.